src/Host/Client/Impl/Session/RSessionRequestSource.cs

```diff
@@ -5,25 +5,26 @@
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Common.Core.Tasks;
 
 namespace Microsoft.R.Host.Client.Session {
     internal sealed class RSessionRequestSource {
```

Comment:
This stuff will benefit from more comments and/or document/diagram explaining the state machine